lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -6451,6 +6452,16 @@ static void inferStaticInitializeObjCMetadata(ClassDecl *classDecl) {
     return;
   }
 
+  // If this class isn't always available on the deployment target, don't
+  // mark it as statically initialized.
+  // FIXME: This is a workaround. The proper solution is for IRGen to
+  // only statically initializae the Objective-C metadata when running on
+  // a new-enough OS.
+  AvailabilityContext availableInfo = AvailabilityContext::alwaysAvailable();
+  if (!tc.isDeclAvailable(classDecl, SourceLoc(),
+                          classDecl->getParentSourceFile(), availableInfo))
```

Comment:
`availableInfo` is an out-parameter; it would make more sense not to initialize it. (I had to go figure out if this was the best way to do what you were trying to do, and this threw me off for a bit.)

Reply:
Yeah, tried that. It has no default initializer, so I have to initialize it with something. *shrug emoji here*